### PR TITLE
Added RST and DTR options to idf-monitor

### PR DESCRIPTION
**Pull request by mmikalsen**
Fixed the issues I experienced with the idf-monitor (#2081), which was related to the RST and DTR pulled lowed by default.

### CODE REVIEW

File: components/esptool_py/Kconfig.projbuild [hunk 1]
```diff
@@ -212,4 +212,11 @@ config MONITOR_BAUD
     default 2000000 if MONITOR_BAUD_2MB
     default MONITOR_BAUD_OTHER_VAL if MONITOR_BAUD_OTHER
 
+config MONITOR_DTR
+		bool "DTR value of idf-monitor"
+
+
+config MONITOR_RST
```
Comment (igrr) [inline components/esptool_py/Kconfig.projbuild]: Here and below: the signal name is RTS ("request to send"), not RST.

**Comment by projectgus** [comment]
Hi @ritualz ,

Thanks for sending this PR! Before we review and merge, I'd like to understand the exact problem that is being solved here:

> which was related to the RST and DTR pulled lowed by default.
...
>-  serial_instance.dtr = False
>-  serial_instance.rts = False

"False" here should mean "not asserted" which for 3.3V serial is usually the same as "not pulled low" (these are active-low signals on every 3.3V serial chip I've come across).

What USB/serial interface are you using, and what wiring configuration to the ESP32 where de-asserting DTR & RTS causes problems?

Thanks,


Angus

**Comment by projectgus** [comment]
@ritualz Apologies, I missed the linked issue. Will take a look there and then reply again.

**Comment by projectgus** [comment]
I found a schematic for the Firebeetle and see that it's a STM32F042 with two GPIOs wired directly to GPIO0 and EN.

There are some soft USB/serial firmwares around which use DTR as the "enable serial" signal, so my guess is that's also what is happening here.

Out of interest, is esptool.py able to flash this board successfully? Does it auto-reset the board into flashing more successfully?

Angus

**Comment by projectgus** [comment]
> Out of interest, is esptool.py able to flash this board successfully? Does it auto-reset the board into flashing more successfully?

Reason I ask this is that while esptool.py is uploading, both DTR and RTS should be de-asserted (set False). I can't explain why this would work and monitor wouldn't work...

File: tools/idf_monitor.py [hunk 2]
```diff
@@ -617,6 +617,22 @@ def main():
         help="Filtering string",
         default=DEFAULT_PRINT_FILTER)
 
+    parser.add_argument(
+        '--dtr',
+        help='Serial port DTR value',
+        const=False,
+        nargs='?',
+        type=bool
```
Comment (weyoui) [inline tools/idf_monitor.py]: I am not sure but feeling we need to add default value for dtr and rts.
```
default = False
```

**Comment by LucaCerina** [comment]
+1 on this pull request. The DF-Robot ESP32 FireBeetle requires a different configuration DTS/RTS to have the monitor working properly

**Comment by chaosman123** [comment]
+1 on this pull request. The DF-Robot ESP32 FireBeetle requires a different configuration DTS/RTS to have the monitor working properly.
Just ran into exactly the same problem. Easy resolved by locally applying the suggested change.

**Pull request closed without merging**